### ISSUE TITLE
groupBy/groupByField: TypeError: t[o.options.groupByField].localeCompare is not a function

### ISSUE THREAD

**Issue by Scnck** [wenzhixin/bootstrap-table#4667]
get error ("TypeError: t[o.options.groupByField].localeCompare is not a function") when attempting to groupByField with number fields suggest fixing logic to to allow number values.

```
	      this.data.sort(function (a, b) {
			return a[that.options.groupByField].toString().localeCompare(b[that.options.groupByField].toString(), 'en-US', { numeric: true });
	      });
```

**Comment by UtechtDustin** [comment]
Please provide us an example using our [Editor](https://live.bootstrap-table.com/).   
Feel free to submit a pull request!

**Comment by Scnck** [comment]
unable to get the error using your editor, but table doesn't fully render using the below code.

https://live.bootstrap-table.com/code/Scnck/999

**Comment by UtechtDustin** [comment]
@Scnck 
The value of the groupByField has to be an string.
Example: https://live.bootstrap-table.com/code/UtechtDustin/1001

I will submit a PR which allows also integers.

**Comment by Scnck** [comment]
I would recommend that you verify if the are really numbers and if so compare as number otherwise as string 
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

// by default, "2" > "10"
console.log("2".localeCompare("10")); // 1

// numeric using options:
console.log("2".localeCompare("10", undefined, {numeric: true})); // -1
```

**Comment by UtechtDustin** [comment]
Thank you for the tip with the numeric option, i added it.
But i dont think we need to check if it is a integer first.

```
Mix of string and integer
[1,44,27, "fasf", "asfasf", "zfsdzdf", "ggsdg"].sort((a, b) => a.toString().localeCompare(b.toString(), undefined, {numeric: true})) //Array(7) [ 1, 27, 44, "asfasf", "fasf", "ggsdg", "zfsdzdf" ]

Only Strings:
["fasf", "asfasf", "zfsdzdf", "ggsdg"].sort((a, b) => a.toString().localeCompare(b.toString())); //Array(4) [ "asfasf", "fasf", "ggsdg", "zfsdzdf" ]

Only Integers:
[1,44,27].sort((a, b) => a.toString().localeCompare(b.toString())); //Array(3) [ 1, 27, 44 ]
```

**Comment by wenzhixin** [comment]
Merged in #4668
